extensions/amp-date-display/0.2/amp-date-display.js

```diff
@@ -0,0 +1,341 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpEvents} from '../../../src/amp-events';
+import {Fragment, createElement} from 'preact';
+import {PreactBaseElement} from '../../../src/preact-base-element';
+import {Services} from '../../../src/services';
+import {createCustomEvent} from '../../../src/event-helper';
+import {dev, devAssert, userAssert} from '../../../src/log';
+import {isExperimentOn} from '../../../src/experiments';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeChildren} from '../../../src/dom';
+import {useResourcesNotify} from '../../../src/preact/utils';
+import {useState} from 'preact/hooks';
+
+/** @const {string} */
+const TAG = 'amp-date-display';
+
+/** @const {string} */
+const DEFAULT_LOCALE = 'en';
+
+/** @const {number} */
+const DEFAULT_OFFSET_SECONDS = 0;
+
+/** @typedef {{
+  year: number,
+  month: number,
+  monthName: string,
+  monthNameShort: string,
+  day: number,
+  dayName: string,
+  dayNameShort: string,
+  hour: number,
+  minute: number,
+  second: number,
+  iso: string,
+}} */
+let VariablesV2Def;
+
+/** @typedef {{
+  year: number,
+  month: number,
+  monthName: string,
+  monthNameShort: string,
+  day: number,
+  dayName: string,
+  dayNameShort: string,
+  hour: number,
+  minute: number,
+  second: number,
+  iso: string,
+  yearTwoDigit: string,
+  monthTwoDigit: string,
+  dayTwoDigit: string,
+  hourTwoDigit: string,
+  hour12: string,
+  hour12TwoDigit: string,
+  minuteTwoDigit: string,
+  secondTwoDigit: string,
+  dayPeriod: string,
+ }} */
+let EnhancedVariablesV2Def;
+
+/**
+ * @param {!JsonObject} props
+ * @return {*} TODO
+ */
+function AmpDateDisplayComponent(props) {
```

Comment:
A nit, but I'd like the React-only stuff to go into a separate module.

Reply:
Done.